lib/unlock-helpers.js

```diff
@@ -0,0 +1,175 @@
+import logger from './logger';
+import { sleep } from 'asyncbox';
+
+const PIN_UNLOCK = "pin";
+const PASSWORD_UNLOCK = "password";
+const PATTERN_UNLOCK = "pattern";
+const UNLOCK_TYPES = [PIN_UNLOCK, PASSWORD_UNLOCK, PATTERN_UNLOCK];
+const KEYCODE_NUMPAD_ENTER = "66";
+
+let helpers = {};
+helpers.isValidUnlockType = function (type) {
+  return UNLOCK_TYPES.indexOf(type) !== -1;
+};
+
+helpers.isValidKey = function (type, key) {
+  if (key === undefined) {
+    return false;
+  }
+  if (type === PIN_UNLOCK) {
+    return /^[0-9]+$/.test(key.trim());
+  }
+  if (type === PATTERN_UNLOCK) {
+    if (!/^[0-9]+$/.test(key.trim())) {
+      return false;
+    }
+    return !(/([0-9]).*?\1/.test(key.trim()));
+  }
+  // Dont trim password key, you can use blank spaces in your android password
+  // ¯\_(ツ)_/¯
+  if (type === PASSWORD_UNLOCK) {
+    return /.{4,}/g.test(key);
+  }
+  throw new Error(`Invalid unlock type ${type}`);
+};
+
+helpers.dismissKeyguard = async function (driver, adb) {
+  let isKeyboardShown = await driver.isKeyboardShown();
+  if (isKeyboardShown) {
+    await driver.hideKeyboard();
+    await sleep(100);
+  }
+  // dismiss notifications
+  await adb.shell(["service", "call", "notification", "1"]);
+  await adb.back();
+  if (await adb.getApiLevel() > 21) {
+    await adb.shell(["wm", "dismiss-keyguard"]);
+  } else {
+    logger.info("swipeUp");
+    await helpers.swipeUp(driver);
+  }
+};
+
+helpers.swipeUp = async function (driver) {
+  let windowSize = await driver.getWindowSize();
+  let x0 = parseInt(windowSize.x/2);
```

Comment:
-> let x0 = parseInt(windowSize.x / 2);
Code style guide says there should be spaces between operands.